### PR TITLE
Use local bundle release override when changing spec

### PR DESCRIPTION
**Pull request by abhay-krishna**
Sometimes, our PRs include changes to spec structs which would also be accompanied by changes in the `config/` folder. During every run of the E2E presubmit, the latest bundle from S3 is pulled down and is parsed to know what images to use and what manifests to apply for cluster creation. One of the steps in cluster creation is installtion of custom components and EKS-A CRDs that are defined in the `eksa-components.yaml` manifest from the bundle release. 

However due to local changes in spec, the eksa-components manifest that was pulled down from S3 will no longer work, since it corresponds to an outdated spec. This PR fixes that by detecting changes under `config/`, that imply that the spec CRD was changed, and generates a local compatible version of the eksa-components to be used in the e2e tests. The tests are configured to run with a local bundle release override in this case, where the eksa-components points to a local file rather an S3 URI.

/hold

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/308#" title="Author self-approved">abhay-krishna</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhay-krishna]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@abhay-krishna: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | 1434961a87f809be3c4dd63e957a3b28de74c7ce | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/308/eks-anywhere-e2e-presubmit/1441664303852687360) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=308). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhay-krishna).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@abhay-krishna: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | 5826dccfedd7b1a2e0e295c238695d916869594e | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/308/eks-anywhere-e2e-presubmit/1442992215444951040) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=308). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhay-krishna).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: scripts/eksa_components_override.sh
Comment (pokearu) [inline scripts/eksa_components_override.sh:39]: I might be wrong here, but this check would work for presubmits but wouldn't the post submits fail?
I mean until the new release-bundle is generated with the new CRDs.

File: test/framework/e2e.go
Comment (pokearu) [inline test/framework/e2e.go:160]: nit: Instead of looking for the file, shall we set bundlesOverride as an env variable during test run? 
Like we do for job ID here
[L97](https://github.com/aws/eks-anywhere/blob/main/internal/test/e2e/setup.go#L97)
Since we are passing it all the way in we might as well set it in the test env.

File: test/framework/e2e.go
Comment (pokearu) [inline test/framework/e2e.go:28]: A bit confused here.
Aren't we running the test from `bin`? Does the `--bundles-override` require an absolute path? 
But i do see that this worked on the presubmit.

Comment (abhay-krishna) [reply test/framework/e2e.go:28]: We're running the test from /home/e2e, and adding /home/e2e/bin to PATH

File: test/framework/e2e.go
Comment (abhay-krishna) [reply test/framework/e2e.go:160]: Sure I can set that. That's better

File: scripts/eksa_components_override.sh
Comment (abhay-krishna) [reply scripts/eksa_components_override.sh:39]: Technically it would, but I've not seen that happen before so left it out. But I can extend this to Codebuild for completeness

**Review by pokearu — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/308#" title="Author self-approved">abhay-krishna</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/308#pullrequestreview-766996357" title="LGTM">pokearu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhay-krishna,pokearu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@abhay-krishna: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-cluster-controller-tooling-presubmit | 90f3ca948705ef2fe06cbea8b607b220d5be24f5 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/308/eks-anywhere-cluster-controller-tooling-presubmit/1443267566133841920) | `/test eks-anywhere-cluster-controller-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=308). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhay-krishna).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
New changes are detected. LGTM label has been removed.

**Review by pokearu — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/308#" title="Author self-approved">abhay-krishna</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/308#pullrequestreview-767050094" title="LGTM">pokearu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhay-krishna,pokearu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
